RaftInvocationFailureTest.test_invocationFailsWitNonRetryableException_when_thereAreRetryableExceptionsAfterIndeterminateOperationState

Jenkins build failed: http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/1429/testReport/junit/com.hazelcast.spi.impl.operationservice.impl/RaftInvocationFailureTest/test_invocationFailsWitNonRetryableException_when_thereAreRetryableExceptionsAfterIndeterminateOperationState/

**Stacktrace**
```
java.lang.AssertionError: com.hazelcast.cp.exception.NotLeaderException: CPMember{uuid=5465ddd3-2451-44c3-8582-175132180a89, address=[127.0.0.1]:5701} is not LEADER of CPGroupId{name='group', seed=0, commitIndex=5}. Known leader is: CPMember{uuid=63c95e60-0746-46cc-9cb6-cae9cee2cdde, address=[127.0.0.1]:5703} is not an instanceof java.lang.IllegalStateException
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.HazelcastTestSupport.assertInstanceOf(HazelcastTestSupport.java:982)
	at com.hazelcast.spi.impl.operationservice.impl.RaftInvocationFailureTest.test_invocationFailsWitNonRetryableException_when_thereAreRetryableExceptionsAfterIndeterminateOperationState(RaftInvocationFailureTest.java:163)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/RaftInvocationFailureTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/RaftInvocationFailureTest.java
@@ -70,6 +70,7 @@ public class RaftInvocationFailureTest extends HazelcastRaftTestSupport {
     public void test_invocationFailsOnMemberLeftException() throws ExecutionException, InterruptedException {
         CPGroupId groupId = getRaftInvocationManager(instances[0]).createRaftGroup(groupName).get();
 
+        waitAllForLeaderElection(instances, groupId);
         HazelcastInstance leader = getLeaderInstance(instances, groupId);
 
         Future f = new RaftInvocation(getOperationServiceImpl(leader).invocationContext,
@@ -90,6 +91,7 @@ public class RaftInvocationFailureTest extends HazelcastRaftTestSupport {
     public void test_invocationFailsWithMemberLeftException_when_thereAreRetryableExceptionsAfterwards() throws ExecutionException, InterruptedException {
         CPGroupId groupId = getRaftInvocationManager(instances[0]).createRaftGroup(groupName).get();
 
+        waitAllForLeaderElection(instances, groupId);
         HazelcastInstance leader = getLeaderInstance(instances, groupId);
 
         Future f = new RaftInvocation(getOperationServiceImpl(leader).invocationContext,
@@ -110,6 +112,7 @@ public class RaftInvocationFailureTest extends HazelcastRaftTestSupport {
     public void test_invocationFailsWithStaleAppendRequestException_when_thereAreRetryableExceptionsAfterwards() throws ExecutionException, InterruptedException {
         CPGroupId groupId = getRaftInvocationManager(instances[0]).createRaftGroup(groupName).get();
 
+        waitAllForLeaderElection(instances, groupId);
         HazelcastInstance leader = getLeaderInstance(instances, groupId);
 
         Future f = new RaftInvocation(getOperationServiceImpl(leader).invocationContext,
@@ -130,6 +133,7 @@ public class RaftInvocationFailureTest extends HazelcastRaftTestSupport {
     public void test_invocationFailsWithFirstMemberLeftException_when_thereAreIndeterminateOperationStateExceptionsAfterwards() throws ExecutionException, InterruptedException {
         CPGroupId groupId = getRaftInvocationManager(instances[0]).createRaftGroup(groupName).get();
 
+        waitAllForLeaderElection(instances, groupId);
         HazelcastInstance leader = getLeaderInstance(instances, groupId);
 
         Future f = new RaftInvocation(getOperationServiceImpl(leader).invocationContext,
@@ -150,6 +154,7 @@ public class RaftInvocationFailureTest extends HazelcastRaftTestSupport {
     public void test_invocationFailsWitNonRetryableException_when_thereAreRetryableExceptionsAfterIndeterminateOperationState() throws ExecutionException, InterruptedException {
         CPGroupId groupId = getRaftInvocationManager(instances[0]).createRaftGroup(groupName).get();
 
+        waitAllForLeaderElection(instances, groupId);
         HazelcastInstance leader = getLeaderInstance(instances, groupId);
 
         Future f = new RaftInvocation(getOperationServiceImpl(leader).invocationContext,

